integration-tests/src/tests/client/features/delegate_action.rs

```diff
@@ -945,3 +952,99 @@ fn meta_tx_create_eth_implicit_account() {
     }
     meta_tx_create_implicit_account(eth_implicit_test_account());
 }
+
+/// Creating an ETH-implicit account with meta-transaction, then attempting to use it with another meta-transaction.
+///
+/// Depending on `rlp_transaction` blob that is sent to the `Wallet Contract`
+/// the transaction is either authorized or unauthorized.
+/// The parameter `authorized` controls which case will be tested.
+fn meta_tx_call_wallet_contract(authorized: bool) {
```

Comment:
Do I understand correctly that the reason to use meta-transactions here is just because of the nature of the placeholder contract not having its own signature checking? So in the future when we have the real Wallet Contract we will be able to remove this test an replace it with tests that directly call the Wallet Contract?

The reason I as is because in the design of the overall solution I do not think we will be using Near meta-transactions. So I want to make sure there is no assumption in the code that meta-transactions are important to the transaction flow.

Reply:
Yes, exactly. I've just added a comment saying that this test should be replaced once Wallet Contract is ready.
There is no assumption about meta-transactions in the code.